mapit_gb/management/commands/mapit_UK_import_ttwa.py

```diff
@@ -0,0 +1,60 @@
+# This script is used to import boundary polygons and other information
+# from shapefiles representing UK travel to work areas
+
+# Run as: ./manage.py mapit_UK_import_ttwa shapefile.shp --commit
+
+from django.core.management.base import LabelCommand
+from django.contrib.gis.gdal import DataSource
+from mapit.models import Area, Generation, Type, NameType, CodeType
+
+
+class Command(LabelCommand):
+    help = 'Creates Travel To Work Area boundaries from Shapefile'
+    label = '<ONS SOA shapefile>'
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+        parser.add_argument('--commit', action='store_true',
+                            dest='commit', help='Actually update the database')
+
+    def handle_label(self, filename, **options):
+        print(filename)
+        generation = Generation.objects.new()
+
+        feat_name = 'TTWA11NM'
+        feat_code = "TTWA11CD"
+        area_type = 'TTW'
+
+        ds = DataSource(filename)
+        layer = ds[0]
+        for feat in layer:
+            # retrieve name and code, and set country
+            name = feat[feat_name].value
+            feature_code = feat[feat_code].value
+
+            # skip if already exists
+            if Area.objects.filter(type__code=area_type, codes__code=feature_code).count():
+                continue
+            print("Adding %s (%s) %s" %
+                  (name, feature_code, feat.geom.geom_name))
+            m = Area(
+                type=Type.objects.get(code=area_type),
+                generation_low=generation,
+                generation_high=generation,
+            )
+            if options['commit']:
+                m.save()
+                m.names.update_or_create(type=NameType.objects.get(
+                    code='S'), defaults={'name': name})
+                m.codes.update_or_create(type=CodeType.objects.get(
+                    code='ons'), defaults={'code': feature_code})
```

Comment:
I think this should be 'gss' which we use for all modern ONS codes (ons is for old SNAC codes).